pandas/tseries/offsets.py

```diff
@@ -177,43 +206,37 @@ def __add__(date):
     """
     _cacheable = False
     _normalize_cache = True
-    _kwds_use_relativedelta = (
-        'years', 'months', 'weeks', 'days',
-        'year', 'month', 'week', 'day', 'weekday',
-        'hour', 'minute', 'second', 'microsecond'
-    )
-    _use_relativedelta = False
     _adjust_dst = False
+    _typ = "dateoffset"
 
-    # default for prior pickles
-    normalize = False
+    def __setattr__(self, name, value):
+        # DateOffset needs to be effectively immutable in order for the
+        # caching in _cached_params to be correct.
+        frozen = ['n', '_offset', 'normalize', '_inc']
+        if name in frozen and hasattr(self, name):
```

Comment:
This PR is not a long-term solution.  Instead how about we have it clear the cache whenever an attribute is re-set?  That should ensure correctness without breaking backward-compatibility.